test/blackbox-tests/test-cases/github5555.t

```diff
@@ -3,145 +3,83 @@ This test is about `binaries` in `env` stanzas in `dune-workspace` files.
-  Called from Fiber__Scheduler.exec in file "src/fiber/scheduler.ml", line 69,
-    characters 8-11
-  Re-raised at Stdune__Exn.raise_with_backtrace in file
-    "otherlibs/stdune/exn.ml", line 36, characters 27-56
-  Called from Fiber__Scheduler.exec in file "src/fiber/scheduler.ml", line 69,
-    characters 8-11
-  Re-raised at Stdune__Exn.raise_with_backtrace in file
-    "otherlibs/stdune/exn.ml", line 36, characters 27-56
-  Called from Fiber__Scheduler.exec in file "src/fiber/scheduler.ml", line 69,
-    characters 8-11
-  Re-raised at Stdune__Exn.raise_with_backtrace in file
-    "otherlibs/stdune/exn.ml", line 36, characters 27-56
-  Called from Fiber__Scheduler.exec in file "src/fiber/scheduler.ml", line 69,
-    characters 8-11
-  Re-raised at Stdune__Exn.raise_with_backtrace in file
-    "otherlibs/stdune/exn.ml", line 36, characters 27-56
-  Called from Fiber__Scheduler.exec in file "src/fiber/scheduler.ml", line 69,
-    characters 8-11
-  Re-raised at Stdune__Exn.raise_with_backtrace in file
-    "otherlibs/stdune/exn.ml", line 36, characters 27-56
-  Called from Fiber__Scheduler.exec in file "src/fiber/scheduler.ml", line 69,
-    characters 8-11
-  Re-raised at Stdune__Exn.raise_with_backtrace in file
-    "otherlibs/stdune/exn.ml", line 36, characters 27-56
-  Called from Fiber__Scheduler.exec in file "src/fiber/scheduler.ml", line 69,
-    characters 8-11
-  -> required by ("<unnamed>", ())
-  -> required by ("<unnamed>", ())
-  -> required by ("<unnamed>", ())
-  -> required by ("<unnamed>", ())
-  -> required by ("<unnamed>", ())
-  -> required by ("<unnamed>", ())
-  -> required by ("load-dir", In_build_dir "default")
-  -> required by ("build-alias", { dir = "default"; name = "default" })
-  -> required by ("toplevel", ())
-  
-  I must not crash.  Uncertainty is the mind-killer. Exceptions are the
-  little-death that brings total obliteration.  I will fully express my cases. 
-  Execution will pass over me and through me.  And when it has gone past, I
-  will unwind the stack along its path.  Where the cases are handled there will
-  be nothing.  Only I will remain.
+  $ t 3.1 dev x.exe
+  File "dune-workspace", line 2, characters 0-28:
+  2 | (env (dev (binaries x.exe)))
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  Error: 'binaries' in an 'env' stanza in a dune-workspace file is only
+  available since version 3.2 of the dune language. Please update your
+  dune-project file to have (lang dune 3.2).
   [1]
 
-When the profile is not selected, this is ignored:
+When the profile is not selected, this is ignored but a warning is printed:
 
-  $ t 3.1 other
+  $ t 3.1 other x.exe
+  File "dune-workspace", line 2, characters 0-30:
+  2 | (env (other (binaries x.exe)))
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  Warning: 'binaries' in an 'env' stanza in a dune-workspace file is only
+  available since version 3.2 of the dune language. Please update your
+  dune-project file to have (lang dune 3.2).
```

Comment:
```suggestion
  available since version 3.2 of the Dune language. Please update your
  dune-project file to have (lang dune 3.2).
```

Reply:
This is the generic "unavailable feature" so I can't fix this one in this PR.